AutomationHelpers/src/UserCodeCollections/WebLibrary.cs

```diff
@@ -89,5 +91,50 @@ public static string GetHttpStatusCode(string url)
 
             return ((int)resp.StatusCode).ToString();
         }
+
+        /// <summary>
+        /// Takes screenshot of entire web page and reports it.
+        /// </summary>
+        /// <param name="repoItemInfo">Repository item</param>
+        [UserCodeMethod]
+        public static void ReportFullPageScreenshot(RepoItemInfo repoItemInfo)
+        {
+            try
+            {
+                if (repoItemInfo == null)
+                {
+                    Report.Error("Repository item must be provided.");
+                    return;
+                }
+
+                WebDocument webDocument = repoItemInfo.CreateAdapter<WebDocument>(false);
+                if (webDocument == null)
+                {
+                    Report.Error("Repository item '" + repoItemInfo.FullName + "' is not web document. " +
+                                 "Screenshot can be taken only for web documents.");
+                    return;
+                }
+
+                var screenshotFilePath = "tmp_screenshot.png";
```

Comment:
IMHO it would be better to create the temporary file really in the temp directory using this method:
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=netframework-4.8

Reply:
Done